### DISCUSSION TITLE
Shuffling location of paired images

### DISCUSSION THREAD

**Discussion by leighgreenberg** [jspsych/jsPsych#2473] [Q&A]
Hello, in my experiment I have image pairs that must be shown on the screen together (e.g. Bob1.jpg and Bob2.jpg). Previously, I  tied the image locations to the image name (e.g. Bob1 always appears on the left, Bob2 always appears on the right), like this:
    

    const image1 = {
        obj_type: 'image',
        file: jsPsych.timelineVariable('stimulus1'),
        show_start_time: 500, // ms after the start of the trial
        startX: 450, //left
    }

    const image2 =  {
        obj_type: 'image',
        file: jsPsych.timelineVariable('stimulus2'),
        show_start_time: 500, // ms after the start of the trial
        startX: 1000 //right 
    }

I'm stuck on how to shuffle the location of the image pairs (i.e. Bob1 could be on the left or the right, and Bob2 would be in the other location) while ensuring the participant still sees the correct pairs. TIA for any insights!

**Comment by jodeleeuw** [comment]
Hi @leighgreenberg,

One way to do this by shuffling the order of the timeline variables so that sometimes `stimulus1` is `Bob1` and sometimes it it is `Bob2`. Is that feasible with your setup?

Another option would be to add a third timeline variable like `stimulus1_left` that is randomly set to `true` or `false`. Then you could change your `file` parameter like this:

```js
file: () => {
  if(jsPsych.timelineVariable('stimulus1_left') == true){
    return jsPsych.timelineVariable('stimulus1'); 
  } else {
    return jsPsych.timelineVariable('stimulus2');
}
```

**Reply by leighgreenberg** [reply]
Hi @jodeleeuw, thank you for taking the time to respond! I couldn't get your suggestions to work for me, I think due to my setup and use of blocks (or my poor understanding of JS..). My solution is definitely not the best but I figured I'd add it here in case anyone has a similar problem in the future: 

I created two more images consts that had the stimulus and location flipped and I paired them (`var locs = [[image1,image2],[image3,image4]];`) then I created a function to randomly choose one of the pairs: 
```JavaScript
var random_combo = function() {
        var image_combo = Math.floor(Math.random() * (1 - 0 + 1)) + 0;
        var final_combo = locs[image_combo];
        return final_combo;
    };
```

Then within the blocks, I set stimuli to `random_combo`.
